grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java

```diff
@@ -41,23 +43,24 @@
 
     private final static Logger LOG = LoggerFactory.getLogger(HALConceptRepresentationBuilder.class);
     private final static int MATCH_QUERY_FIXED_DEGREE = 0;
-    private final static String ASSERTION_URL = REST.WebPath.GRAPH_MATCH_QUERY_URI + "?query=match $x id '%s'; $y id '%s'; $r (%s$x, %s$y); select $r;";
+    private final static String ASSERTION_URL = REST.WebPath.GRAPH_MATCH_QUERY_URI + "?query=match $x id '%s'; $y id '%s'; $r (%s$x, %s$y) %s; select $r;";
     private final static String HAS_ROLE_EDGE = "EMPTY-GRAKN-ROLE";
 
     public static Json renderHALArrayData(MatchQuery matchQuery, Collection<Map<String, Concept>> graqlResultsList, String keyspace) {
+
         //Compute connections between variables in Graql result
-        Map<String, Collection<String>> linkedNodes = new HashMap<>();
-        Map<String, String> roleTypes = new HashMap<>();
+        Map<String, Collection<VarAdmin>> linkedNodes = new HashMap<>();
+        Map<String,Map<String, String>> roleTypes = new HashMap<>();
```

Comment:
I am guessing `roleTypes` and `linkedNodes` are filled in `computeLinkedNodesFromQuery` ? That feels odd I would say it should be something like: `roleTypes = compute....` and `linkedNodes = compute....`. I.e. it should be set through the return type ideally. This is really nit picky though.